framework/src/main/java/org/tron/core/net/service/effective/ResilienceService.java

```diff
@@ -0,0 +1,168 @@
+package org.tron.core.net.service.effective;
+
+import java.util.Comparator;
+import java.util.List;
+import java.util.Optional;
+import java.util.Random;
+import java.util.concurrent.ScheduledExecutorService;
+import java.util.concurrent.TimeUnit;
+import java.util.stream.Collectors;
+import lombok.extern.slf4j.Slf4j;
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.stereotype.Component;
+import org.tron.common.es.ExecutorServiceManager;
+import org.tron.common.parameter.CommonParameter;
+import org.tron.core.ChainBaseManager;
+import org.tron.core.config.args.Args;
+import org.tron.core.net.TronNetDelegate;
+import org.tron.core.net.peer.PeerConnection;
+import org.tron.protos.Protocol.ReasonCode;
+
+@Slf4j(topic = "net")
+@Component
+public class ResilienceService {
+
+  private static final long inactiveThreshold =
+      CommonParameter.getInstance().getInactiveThreshold() * 1000L;
+  public static final long blockNotChangeThreshold = 90 * 1000L;
+
+  //when node is isolated, retention percent peers will not be disconnected
+  public static final double retentionPercent = 0.8;
+  private static final int initialDelay = 300;
+  private static final String esName = "resilience-service";
+  private final ScheduledExecutorService executor = ExecutorServiceManager
+      .newSingleThreadScheduledExecutor(esName);
+
+  @Autowired
+  private TronNetDelegate tronNetDelegate;
+
+  @Autowired
+  private ChainBaseManager chainBaseManager;
+
+  public void init() {
+    if (Args.getInstance().isOpenFullTcpDisconnect) {
+      executor.scheduleWithFixedDelay(() -> {
+        try {
+          disconnectRandom();
+        } catch (Exception e) {
+          logger.error("DisconnectRandom node failed", e);
+        }
+      }, initialDelay, 60, TimeUnit.SECONDS);
+    } else {
+      logger.info("OpenFullTcpDisconnect is disabled");
+    }
+
+    executor.scheduleWithFixedDelay(() -> {
+      try {
+        disconnectLan();
+      } catch (Exception e) {
+        logger.error("DisconnectLan node failed", e);
+      }
+    }, initialDelay, 10, TimeUnit.SECONDS);
+
+    executor.scheduleWithFixedDelay(() -> {
+      try {
+        disconnectIsolated2();
+      } catch (Exception e) {
+        logger.error("DisconnectIsolated node failed", e);
+      }
+    }, initialDelay, 30, TimeUnit.SECONDS);
+  }
+
+  private void disconnectRandom() {
+    int peerSize = tronNetDelegate.getActivePeer().size();
+    if (peerSize >= CommonParameter.getInstance().getMaxConnections()) {
+      long now = System.currentTimeMillis();
+      List<PeerConnection> peers = tronNetDelegate.getActivePeer().stream()
+          .filter(peer -> !peer.isDisconnect())
+          .filter(peer -> now - peer.getLastActiveTime() >= inactiveThreshold)
+          .filter(peer -> !peer.getChannel().isTrustPeer())
```

Comment:
Why change the original logic, not excluding active connections.

Reply:
Original logic don't have the inactiveThreshold restrict, it has many conditates. The number of peers whose inactive time is more inactiveThreshold is very small, it has very little condidates, so there is no need to exclude active connections.